From b89a509acec0dec627929f998971385b49331493 Mon Sep 17 00:00:00 2001
From: Henri Yandell <bayard@apache.org>
Date: Tue, 25 Mar 2008 07:52:27 +0000
Subject: [PATCH] Removed the recursive method call, and pointed the
 read(byte[]) to the read(byte[],int,int) method as per VFS-190

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/vfs/trunk@640732 13f79535-47bb-0310-9956-ffa450edef68
---
 .../commons/vfs/provider/ram/RamFileRandomAccessContent.java   | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/core/src/main/java/org/apache/commons/vfs/provider/ram/RamFileRandomAccessContent.java b/core/src/main/java/org/apache/commons/vfs/provider/ram/RamFileRandomAccessContent.java
index 39ae79f266..5d53c61af9 100644
--- a/core/src/main/java/org/apache/commons/vfs/provider/ram/RamFileRandomAccessContent.java
+++ b/core/src/main/java/org/apache/commons/vfs/provider/ram/RamFileRandomAccessContent.java
@@ -105,12 +105,11 @@ public long skip(long n) throws IOException
 
 			public void close() throws IOException
 			{
-				close();
 			}
 
 			public int read(byte b[]) throws IOException
 			{
-				return read(b);
+				return read(b, 0, b.length);
 			}
 
 			public int read(byte b[], int off, int len) throws IOException
